src/js/subtraction/components/Subtraction.tsx

```diff
@@ -12,11 +12,11 @@ export default function Subtraction() {
     return (
         <Container>
             <ContainerNarrow>
-                <Switch>
-                    <Route path="/subtractions" component={SubtractionList} exact />
-                    <Route path="/subtractions/files" component={SubtractionFileManager} />
-                    <Route path="/subtractions/:subtractionId" component={SubtractionDetail} />
-                </Switch>
+                <Routes>
+                    <Route path="/subtractions" element={<SubtractionList />} />
+                    <Route path="/subtractions/files" element={<SubtractionFileManager />} />
+                    <Route path="/subtractions/:subtractionId" element={<SubtractionDetail />} />
+                </Routes>
```

Comment:
On a glance I think the test failure is a path mismatch problem. Currently when directly rendering `Subtraction`, react router thinks that the URL of the details page is `/subtractions/:subtractionId` while in testing the URL being passed to MemoryRouter is `/:subtractionId`.

To resolve I would do two things:

1. In main update the `Route` used for subtratcions to `CompatRoute` with path `/subtractions`. (use samples as an example)
2. Update these paths to use the relative addressing

I think doing both should resolve the testing problems

Reply:
Good catch!